src/reconcile/rec_visibility.c

```diff
@@ -49,6 +49,11 @@ __rec_update_save(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_INSERT *ins, WT
         r->supd_memsize += upd_memsize;
     } else
         WT_ASSERT(session, !F_ISSET(r, WT_REC_EVICT) || upd_memsize == 0);
+
+    if (F_ISSET(S2BT(session), WT_BTREE_DISAGGREGATED) && !r->none_durable_upd_used &&
```

Comment:
Would it be possible to add some kind of test for this? If it fails, it'd be extremely hard to tell where the bug is.